packages/core/src/internal/execution/jsonrpc-client.ts

```diff
@@ -674,6 +679,28 @@ export class EIP1193JsonRpcClient implements JsonRpcClient {
       };
     }
 
+    /**
+     * Polygon amoy testnet (chainId 80002) currently has a bug causing the
+     * `eth_gasPrice` RPC call to return an amount that is way too high.
+     * We hardcode the gas price for this chain for now until the bug is fixed.
+     * See: https://github.com/maticnetwork/bor/issues/1213
+     *
+     * Note that at the time of this implementation, the issue was autoclosed by a bot
+     * as a maintainer had not responded to the issue yet. Users continue to report
+     * the bug in the issue comments, however.
+     *
+     * All of that to say, when evaluating whether this logic is still needed in the future,
+     * it will likely be required to read through the issue above, rather than relying on the
+     * status of the github issue itself.
```

Comment:
This is great!

I mean, your comment helping our future selves, the bug is so sad 🤣